[iOS] Silence Network process denial to read /private/etc/group

brentfulgham opened the pull request:
#### 3b1e349c8e579c0a0af3a21bb2267322bf3f6ff8
<pre>
[iOS] Silence Network process denial to read /private/etc/group
<a href="https://bugs.webkit.org/show_bug.cgi?id=276560">https://bugs.webkit.org/show_bug.cgi?id=276560</a>
&lt;<a href="https://rdar.apple.com/problem/131639214">rdar://problem/131639214</a>&gt;

Reviewed by Per Arne Vollan.

I noticed that the iOS Network process generates a sandbox violation attempting to read
`/private/etc/group` every three seconds while playing video.

Group doesn&apos;t contain anything interesting on iOS, and is not relevant to any WebKit use
cases. We should silence this to avoid the telemetry overhead.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:

Canonical link: <a href="https://commits.webkit.org/280983@main">https://commits.webkit.org/280983@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/30ff02876446c0a00b160c047917c6c283561a12

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/58221 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37549 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10704 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61846 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8666 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/45185 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8860 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/47181 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6194 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/60252 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35202 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/50339 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28014 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31965 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7670 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53920 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7905 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63551 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/2135 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7966 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54500 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/2143 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/50351 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54555 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12877 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1829 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/33378 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34464 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35548 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/34209 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pvollan reviewed (approved):
R=me.

pvollan commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:
```diff
@@ -374,7 +374,8 @@
 
 (define required-etc-files
-  (literal "/private/etc/hosts"
+  (literal "/private/etc/group"
```
I think we should also consider whether we would like to silence this, since it doesn't seem necessary for a Networking process to have knowledge about group information.

brentfulgham replied:
That's a good point. I looked through `libinfo` use cases, and can't see where it's accessed in a way that's relevant to WebKit uses. I'll revise this patch to silence the warning.

I did an A/B performance test to see if allowing it changed anything, and there were no improvements so there doesn't seem to be any value in unblocking.

pvollan replied:
Thank you!

brentfulgham committed:
[iOS] Silence Network process denial to read /private/etc/group
https://bugs.webkit.org/show_bug.cgi?id=276560
<rdar://problem/131639214>

Reviewed by Per Arne Vollan.

I noticed that the iOS Network process generates a sandbox violation attempting to read
`/private/etc/group` every three seconds while playing video.

Group doesn't contain anything interesting on iOS, and is not relevant to any WebKit use
cases. We should silence this to avoid the telemetry overhead.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.Networking.sb.in:

Canonical link: https://commits.webkit.org/280983@main
